handbook/engineering/web/index.md

```diff
@@ -13,7 +13,20 @@ The web team's current focus is documented in [the tracking issue for the curren
 
 ## Goals
 
-TODO
+### Long term
+
+***Deliver our users the full, unique value of [extensions](https://docs.sourcegraph.com/extensions).***
+
+**Outcome**: Our webapp, browser extensions and native integrations are great platforms to provide our users unique value through extensions. These platforms are well-maintained, consistent, easy to setup, well-documented, well-tested, performant, and show their power through convincing extensions built on top of them. The extensions platform and API provide powerful capabilities to extension developers and a great developer experience.
+
+### Medium term
+
+To reach our long-term goal, we set the following medium-term goals to guide our short-term iteration plans.
+We will gradually transition our focus through these goals in the order they are defined here.
```

Comment:
Felix and I discussed that these are not hard incremental boundaries, and work will be done in order, but there will be some overlap.

```suggestion
We will tackle these medium-term goals in order, though expect to have some work done in parallel as we progress. 
```

Reply:
Yeah, that's what I actually already meant here with "gradually", thanks for making that clearer!